designs/odcr.md

```diff
@@ -0,0 +1,121 @@
+# On demand capacity reservation
+
+This documents proposes supporting ODCR in Karpenter
+
+## Background
+
+In AWS [ODCR](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-capacity-reservations.html) allows users to reserve compute capacity to mitigate against the risk of being 
+unabled to get on demand capacity. This is very helpful during seasonal holidays where higher traffic is expected. 
+
+### Capacity Reservations
+
+Each Capacity Reservation is defined with:
+
+- The Availability Zone in which to reserve the capacity
+- The number of instances for which to reserve capacity
+- The instance attributes, including the instance type, tenancy, and platform/OS
+- Instance match criteria
+  - If Capacity Reservation is targeted, only accept instances that matches all attributes + explicitly targeted the capacity reservation
+  - Open if capacity reservation accepts all instances that matches all attributres
+
+AWS also supports grouping Capacity Reservation into Capacity Reservation groups. 
+Both these entities are supported in Launch Template's CapacityReservationTarget [definitions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-capacityreservationtarget.html).
+
+## Goals
+
+- Support associating ODCR to EC2NodeClass
+- Define Karpenter's behavior when launching nodes into Capacity Reservation
+- Define Karpenter's behavior when encountering errors when attempting to launch nodes into Capacity Reservation
+
+## Non-Goals
+
+_We are keeping the scope of this design very targeted so even if these could be things we eventually support, we aren't scoping them into this design_
+- Supporting prioritization when launching nodes 
```

Comment:
Doesn't Karpenter already support this through its NodePool "weight" concept?

Reply:
I meant within a single node pool. I can update